lib/elf/file.rb

```diff
@@ -374,12 +374,16 @@ def is_compatible(other)
       raise TypeError.new("wrong argument type #{other.class} (expected Elf::File)") unless
         other.is_a? Elf::File
 
+      compatible_abi = case
+        when [@abi, other.abi].include?(Elf::OsAbi::SysV); true
```

Comment:
I did changes based on [one of the responses on StackOverflow for question "Why does readelf show “System V” as my OS instead of Linux?"](http://unix.stackexchange.com/a/132119/11448) which says that SysV means that ELF file have no OS specific ABI.

I need to find better one. Maybe I need to check ABI version as well. But unfortunately at this moment I can't find any reference that describes compatibility between ELFs with different OS ABI values.

As a fall-back I can write here exact case I have:
```
zsh% file -L =lyx
/usr/sbin/lyx: ELF 64-bit LSB executable, x86-64, version 1 (GNU/Linux), dynamically linked, interpreter /usr/x86_64-pc-linux-gnu/lib/ld-linux-x86-64.so.2, for GNU/Linux 2.6.34, BuildID[sha1]=b218bb46b809a99108cae1dc55ee95a3b63e9570, stripped
zsh% ldd =lyx | grep QtGui
	libQtGui.so.4 => /usr/x86_64-pc-linux-gnu/lib/qt4/libQtGui.so.4 (0x00007fba2b91d000)
zsh% file -L /usr/x86_64-pc-linux-gnu/lib/qt4/libQtGui.so.4
/usr/x86_64-pc-linux-gnu/lib/qt4/libQtGui.so.4: ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, stripped
```

And probably `#is_compatible` will be asymmetric in this case.